From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: thomasclark <tjclark@wpi.edu>
Date: Wed, 14 May 2014 14:23:39 -0400
Subject: [PATCH 0049/6262] Revert "Modifies maven files to work with m2e" This
 commit causes Eclipse to give a vaugue error message when creating a new
 Robot Java Project or Robot C++ Project.

This reverts commit 36241b0d6c79a054733939a413fcecf6355d0147.

Change-Id: I47b66c53604f07b6e40c0db6aa250b977d9ae903
---
 .../.project                                  | 17 ++++++
 .../.classpath                                |  7 +++
 .../.project                                  | 28 +++++++++
 .../.project                                  | 17 ++++++
 .../.project                                  | 17 ++++++
 .../.classpath                                |  7 +++
 .../.project                                  | 28 +++++++++
 .../.project                                  | 17 ++++++
 .../.classpath                                |  7 +++
 .../.project                                  | 28 +++++++++
 .../.project                                  | 17 ++++++
 .../.classpath                                |  7 +++
 .../.project                                  | 28 +++++++++
 .../.classpath                                |  7 +++
 .../edu.wpi.first.wpilib.plugins.cpp/.project | 28 +++++++++
 .../edu.wpi.first.wpilib.plugins.cpp/pom.xml  |  2 -
 .../.project                                  | 17 ++++++
 .../.classpath                                |  7 +++
 .../.project                                  | 34 +++++++++++
 .../edu.wpi.first.wpilib.plugins.java/pom.xml |  2 -
 .../.project                                  | 17 ++++++
 pom.xml                                       | 60 +++++++++----------
 wpilibj/wpilibJavaFinal/pom.xml               |  2 -
 23 files changed, 364 insertions(+), 37 deletions(-)
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.core.feature/.project
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.core/.classpath
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.core/.project
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.feature/.project
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.linux.feature/.project
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.linux/.classpath
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.linux/.project
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.mac.feature/.project
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.mac/.classpath
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.mac/.project
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.windows.feature/.project
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.windows/.classpath
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.windows/.project
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/.classpath
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/.project
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.java.feature/.project
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.java/.classpath
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.java/.project
 create mode 100644 eclipse-plugins/edu.wpi.first.wpilib.plugins.updatesite/.project

diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.core.feature/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.core.feature/.project
new file mode 100644
index 0000000000000000000000000000000000000000..ade1d91b15a3ba2ba62e19ae190f172d9f928e4a
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.core.feature/.project
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.core.feature</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.pde.FeatureBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.FeatureNature</nature>
+	</natures>
+</projectDescription>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.core/.classpath b/eclipse-plugins/edu.wpi.first.wpilib.plugins.core/.classpath
new file mode 100644
index 0000000000000000000000000000000000000000..c0b4c7e1122f042f36cbb1fb30421814da72ce9d
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.core/.classpath
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7"/>
+	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
+	<classpathentry kind="src" path="src/main/java"/>
+	<classpathentry kind="output" path="bin"/>
+</classpath>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.core/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.core/.project
new file mode 100644
index 0000000000000000000000000000000000000000..10ca0d4f0395ba2c3ed79978b03e79245edc1e21
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.core/.project
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.core</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.ManifestBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.PluginNature</nature>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+	</natures>
+</projectDescription>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.feature/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.feature/.project
new file mode 100644
index 0000000000000000000000000000000000000000..cb3157daa3bf62ad75dcd93fc9760470b2612e98
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.feature/.project
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.cpp.feature</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.pde.FeatureBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.FeatureNature</nature>
+	</natures>
+</projectDescription>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.linux.feature/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.linux.feature/.project
new file mode 100644
index 0000000000000000000000000000000000000000..d7ffa026d485b0e4c3e474f6bf551bebaeef62d3
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.linux.feature/.project
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.cpp.toolchains.linux.feature</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.pde.FeatureBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.FeatureNature</nature>
+	</natures>
+</projectDescription>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.linux/.classpath b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.linux/.classpath
new file mode 100644
index 0000000000000000000000000000000000000000..098194ca4b7d8f45177f94e735506ae3a26b5c94
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.linux/.classpath
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7"/>
+	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="output" path="bin"/>
+</classpath>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.linux/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.linux/.project
new file mode 100644
index 0000000000000000000000000000000000000000..8317f12f59f29cc26e74838a574fd80189573968
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.linux/.project
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.cpp.toolchains.linux</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.ManifestBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.PluginNature</nature>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+	</natures>
+</projectDescription>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.mac.feature/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.mac.feature/.project
new file mode 100644
index 0000000000000000000000000000000000000000..eedadaa4a77fdd5b68d55e4de6a053269a2bb682
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.mac.feature/.project
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.cpp.toolchains.mac.feature</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.pde.FeatureBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.FeatureNature</nature>
+	</natures>
+</projectDescription>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.mac/.classpath b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.mac/.classpath
new file mode 100644
index 0000000000000000000000000000000000000000..098194ca4b7d8f45177f94e735506ae3a26b5c94
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.mac/.classpath
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7"/>
+	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="output" path="bin"/>
+</classpath>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.mac/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.mac/.project
new file mode 100644
index 0000000000000000000000000000000000000000..bb3deccb96c078164a9458755f77964f69d93287
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.mac/.project
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.cpp.toolchains.mac</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.ManifestBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.PluginNature</nature>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+	</natures>
+</projectDescription>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.windows.feature/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.windows.feature/.project
new file mode 100644
index 0000000000000000000000000000000000000000..d7a19269543449f51225880e4b2f364e3bc6f242
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.windows.feature/.project
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.cpp.toolchains.windows.feature</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.pde.FeatureBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.FeatureNature</nature>
+	</natures>
+</projectDescription>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.windows/.classpath b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.windows/.classpath
new file mode 100644
index 0000000000000000000000000000000000000000..098194ca4b7d8f45177f94e735506ae3a26b5c94
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.windows/.classpath
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7"/>
+	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="output" path="bin"/>
+</classpath>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.windows/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.windows/.project
new file mode 100644
index 0000000000000000000000000000000000000000..4b17efb32b71535e50911d669b54600b7959c09c
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp.toolchains.windows/.project
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.cpp.toolchains.windows</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.ManifestBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.PluginNature</nature>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+	</natures>
+</projectDescription>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/.classpath b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/.classpath
new file mode 100644
index 0000000000000000000000000000000000000000..c0b4c7e1122f042f36cbb1fb30421814da72ce9d
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/.classpath
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7"/>
+	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
+	<classpathentry kind="src" path="src/main/java"/>
+	<classpathentry kind="output" path="bin"/>
+</classpath>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/.project
new file mode 100644
index 0000000000000000000000000000000000000000..325460c3af26d23f3a76cfe268731ab51aab425d
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/.project
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.cpp</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.ManifestBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.PluginNature</nature>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+	</natures>
+</projectDescription>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/pom.xml b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/pom.xml
index 4f00e6d358402a3993a92429a9bf78e9e0b0d1fe..c74a120789bb7ed3a81aa5ec13578e56b3d919b6 100644
--- a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/pom.xml
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/pom.xml
@@ -23,7 +23,6 @@
 				<filtering>true</filtering>
 			</resource>
 		</resources>
-		<pluginManagement>
 		<plugins>
 			<plugin>
 				<artifactId>maven-resources-plugin</artifactId>
@@ -134,7 +133,6 @@
 				</executions>
 			</plugin>
 		</plugins>
-		</pluginManagement>
 	</build>
 	<dependencies>
 		<dependency>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.java.feature/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.java.feature/.project
new file mode 100644
index 0000000000000000000000000000000000000000..bd4257bb4d70ad4e9df5469bccddf2adb913638c
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.java.feature/.project
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.java.feature</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.pde.FeatureBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.FeatureNature</nature>
+	</natures>
+</projectDescription>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.java/.classpath b/eclipse-plugins/edu.wpi.first.wpilib.plugins.java/.classpath
new file mode 100644
index 0000000000000000000000000000000000000000..c0b4c7e1122f042f36cbb1fb30421814da72ce9d
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.java/.classpath
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7"/>
+	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
+	<classpathentry kind="src" path="src/main/java"/>
+	<classpathentry kind="output" path="bin"/>
+</classpath>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.java/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.java/.project
new file mode 100644
index 0000000000000000000000000000000000000000..02b92fbf3402769c73b97569a2b6636cbd5dea4f
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.java/.project
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.java</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.ManifestBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
+		<nature>org.eclipse.pde.PluginNature</nature>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+	</natures>
+</projectDescription>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.java/pom.xml b/eclipse-plugins/edu.wpi.first.wpilib.plugins.java/pom.xml
index 5e668e56c571b672728cb9cbef28a56c98c90069..148b79fb985391736d87b3fe61779b66d61c7e38 100644
--- a/eclipse-plugins/edu.wpi.first.wpilib.plugins.java/pom.xml
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.java/pom.xml
@@ -40,7 +40,6 @@
 		<filtering>true</filtering>
       </resource>
 	</resources>
-	<pluginManagement>
 	<plugins>
 	  <plugin>
         <artifactId>maven-resources-plugin</artifactId>
@@ -207,7 +206,6 @@
 		</executions>
 	  </plugin>
     </plugins>
-	</pluginManagement>
   </build>
   
   <dependencies>
diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.updatesite/.project b/eclipse-plugins/edu.wpi.first.wpilib.plugins.updatesite/.project
new file mode 100644
index 0000000000000000000000000000000000000000..0bbfd83d3fb8c91a8cab85b95e5d09918a988084
--- /dev/null
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.updatesite/.project
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>edu.wpi.first.wpilib.plugins.updatesite</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.pde.UpdateSiteBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.UpdateSiteNature</nature>
+	</natures>
+</projectDescription>
diff --git a/pom.xml b/pom.xml
index ce3bf52c0f4f937c91180f7917a0712bfe453371..bff1e228325ead06e217a24cf7ba23e274c904d5 100644
--- a/pom.xml
+++ b/pom.xml
@@ -14,36 +14,34 @@
 		<module>driver-station</module>
 	</modules>
 	<build>
-		<pluginManagement>
-			<plugins>
-				<plugin>
-					<groupId>org.apache.maven.plugins</groupId>
-					<artifactId>maven-enforcer-plugin</artifactId>
-					<version>1.3.1</version>
-					<executions>
-						<execution>
-							<id>enforce-maven</id>
-							<goals>
-								<goal>enforce</goal>
-							</goals>
-							<configuration>
-								<rules>
-									<requireMavenVersion>
-										<!-- Require Maven version > 3.0.0, but exclude
-	                  3.1.0 due to http://jira.codehaus.org/browse/MNG-5503
-	                  -->
-										<version>[3.0.0,3.1.0), (3.1.0,)</version>
-										<message>WPILib has been tested with Maven 3.0.0+. 
-	Note: Maven 3.1.0 is not supported due to Maven bug MNG-5503 (http://jira.codehaus.org/browse/MNG-5503). Please upgrade
-	to Maven 3.1.1 or later.
-	                  </message>
-									</requireMavenVersion>
-								</rules>
-							</configuration>
-						</execution>
-					</executions>
-				</plugin>
-			</plugins>
-		</pluginManagement>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-enforcer-plugin</artifactId>
+				<version>1.3.1</version>
+				<executions>
+					<execution>
+						<id>enforce-maven</id>
+						<goals>
+							<goal>enforce</goal>
+						</goals>
+						<configuration>
+							<rules>
+								<requireMavenVersion>
+									<!-- Require Maven version > 3.0.0, but exclude
+                  3.1.0 due to http://jira.codehaus.org/browse/MNG-5503
+                  -->
+									<version>[3.0.0,3.1.0), (3.1.0,)</version>
+									<message>WPILib has been tested with Maven 3.0.0+. 
+Note: Maven 3.1.0 is not supported due to Maven bug MNG-5503 (http://jira.codehaus.org/browse/MNG-5503). Please upgrade
+to Maven 3.1.1 or later.
+                  </message>
+								</requireMavenVersion>
+							</rules>
+						</configuration>
+					</execution>
+				</executions>
+			</plugin>
+		</plugins>
 	</build>
 </project>
diff --git a/wpilibj/wpilibJavaFinal/pom.xml b/wpilibj/wpilibJavaFinal/pom.xml
index 028de650e7a035753501ddde0fee15c2a0fa942b..a2bf252b9b2943d62fddc37930a1bfee259e6caf 100644
--- a/wpilibj/wpilibJavaFinal/pom.xml
+++ b/wpilibj/wpilibJavaFinal/pom.xml
@@ -42,7 +42,6 @@
   </dependencies>
 
   <build>
-  <pluginManagement>
 	<plugins>
 	  <plugin>
         <groupId>org.apache.maven.plugins</groupId>
@@ -113,6 +112,5 @@
       </executions>
       </plugin>
 	</plugins>
-  </pluginManagement>
   </build>
 </project>
